packages/desktop/src/main/main.ts

```diff
@@ -371,6 +371,21 @@ app.on('ready', async () => {
     closeBackendProcess()
   })
 
+  ipcMain.on('writeTempFile', (event, arg) => {
+    const temporaryFilesDirectory = path.join(appDataPath, 'temporaryFiles')
+    fs.mkdirSync(temporaryFilesDirectory, { recursive: true })
+    const filePath = `${path.join(temporaryFilesDirectory, arg.fileName)}`
+    const writeStream = fs.createWriteStream(filePath)
```

Comment:
Since we have the complete data here we can probably use more appropriate method here eg. writeFile